lib/rucio/core/rule.py

```diff
@@ -1747,7 +2027,7 @@ def get_updated_dids(total_workers: int, worker_number: int, limit: int = 100, b
         else:
             return filtered_dids
     else:
-        return [did for did in query.order_by(models.UpdatedDID.created_at).all() if (did.scope, did.name) not in blocked_dids]
+        return [did for did in session.execute(stmt.order_by(models.UpdatedDID.created_at)).all() if (did.scope, did.name) not in blocked_dids]  # type: ignore
```

Comment:
I don’t understand how the hint for the return type is `list[tuple[str, str]]`. I think putting `#type: ignore` is the wrong approach.

I’m afraid that #6516 may have introduced some mistakes.

This comment applies throughout the file.